pkg/core/message.go

```diff
@@ -110,9 +112,10 @@ type Message struct {
 // Fields such as the state/confirmed do NOT transfer, as these are calculated individually by each member.
 func (m *Message) BatchMessage() *Message {
 	return &Message{
-		Header: m.Header,
-		Hash:   m.Hash,
-		Data:   m.Data,
+		Header:   m.Header,
+		Hash:     m.Hash,
+		Data:     m.Data,
+		TxParent: m.TxParent, // TODO: this should really move inside Header
```

Comment:
Important TODO here - I think this should be possible, but requires some re-shuffling of logic when building the token transfer/approval, so that the transaction ID is computed before the message is sealed.